### DISCUSSION TITLE
Help with build error on vercel

### DISCUSSION THREAD

**Discussion by waptik** [vercel/next.js#17168] [Help] [answered]
Hi! I'm having some issues when my app is going through the build process on Vercel.
I am using typescript to set path aliases. When running `yarn build` locally which has `next build`, the build is successful without any issues. But my push my code to github and vercel is trying to build it, i get this error 
```sh
09:12:44.668 | Failed to compile.
-- | --
09:12:44.668 | ModuleNotFoundError: Module not found: Error: Can't resolve '@/components/header' in '/vercel/5bdbbc1f/pages'
09:12:44.668 | > Build error occurred
09:12:44.669 | Error: > Build failed because of webpack errors
09:12:44.669 | at build (/vercel/5bdbbc1f/node_modules/next/dist/build/index.js:15:918)
09:12:44.669 | at runMicrotasks (<anonymous>)
09:12:44.669 | at processTicksAndRejections (internal/process/task_queues.js:97:5)
09:12:44.695 | error Command failed with exit code 1.
```

Here's how i set my path baseurl and path aliases:
```ts
...
"baseUrl": ".",
      "paths": {
        "@/components/*": ["components/*"],
        "@/hooks/*": ["hooks/*"],
        "@/lib/*": ["lib/*"],
        "@/styles/*": ["styles/*"],
        "@/utils/*": ["utils/*"]
      },
...
```
I was previously using this version of nextjs `^9.5.4-canary.3` but after seeing issue #17095, i updated it to version `^9.5.4-canary.20` but that didn't solve it.

**Comment by leerob** [comment]
Can you try using a non-canary version like `9.5.3`?

**Comment by waptik** [comment, marked as answer]
Thanks for the suggestion but that didn't help.
`header` is not a file but rather folder described as follows:
```js
/header
  - index.tsx
  - TopNavBar.tsx
  - TopNavBarProfileMenu.tsx
```
Btw, i disabled typescript build error in `next.config.js` but it didn't solve it.
I have other component folders with similar structure. I import header in other files that need it like this: `import Header from '@/components/header`. After trying your suggestion and didn't work, i thought maybe the issue was because the folder name was in lowercase so i changed it to `Header` and changed the name accordingly everywhere it was being imported(`import Header from '@/components/Header`). I think that solved the issue.

**Reply by withoutwax** [reply]
I had a same issue, and it was the case-sensitive naming of the files that caused the issue.

**Reply by tsvecak** [reply]
I had same issue by my git GUI didn't recognise the case-sensitive change so I've just removed those files, commit, added back those files(with correct casing), commit and it works 👍
